<?xml version="1.0" ?>

<launch>

    <node pkg="joy" type="joy_node" name="joy_node">
        <param name="autorepeat_rate" value="20" />
    </node>

    <include file="$(find subt_rl)/launch/pcToLaser_240_duckieboat.launch">
        <arg name="veh" value="boat1" />
    </include>
    <group ns="boat1">
        <node pkg="subt_rl" type="teleop.py" name="teleop" />
        <node pkg="subt_rl" type="torch_rdpg_forward_duckieboat.py" name="run_rdpg_duckieboat" output="screen">
            <param name="model" value="s1536_f1869509.pth" />
            <remap from="laser" to="RL/scan" />
            <remap from="odom" to="p3d_odom" />
            <remap from="cmd_vel" to="cmd_vel" />
            <remap from="joy" to="/joy" />
        </node>
    </group>

    <include file="$(find subt_rl)/launch/pcToLaser_240_duckieboat.launch">
        <arg name="veh" value="boat2" />
    </include>
    <group ns="boat2">
        <node pkg="subt_rl" type="teleop.py" name="teleop" />
        <node pkg="subt_rl" type="torch_rdpg_forward_duckieboat.py" name="run_rdpg_duckieboat" output="screen">
            <param name="model" value="s1536_f1869509.pth" />
            <remap from="laser" to="RL/scan" />
            <remap from="odom" to="p3d_odom" />
            <remap from="cmd_vel" to="cmd_vel" />
            <remap from="joy" to="/joy" />
        </node>
    </group>

    <include file="$(find subt_rl)/launch/pcToLaser_240_duckieboat.launch">
        <arg name="veh" value="boat3" />
    </include>
    <group ns="boat3">
        <node pkg="subt_rl" type="teleop.py" name="teleop" />
        <node pkg="subt_rl" type="torch_rdpg_forward_duckieboat.py" name="run_rdpg_duckieboat" output="screen">
            <param name="model" value="s1536_f1869509.pth" />
            <remap from="laser" to="RL/scan" />
            <remap from="odom" to="p3d_odom" />
            <remap from="cmd_vel" to="cmd_vel" />
            <remap from="joy" to="/joy" />
        </node>
    </group>

    <include file="$(find subt_rl)/launch/pcToLaser_240_duckieboat.launch">
        <arg name="veh" value="boat4" />
    </include>
    <group ns="boat4">
        <node pkg="subt_rl" type="teleop.py" name="teleop" />
        <node pkg="subt_rl" type="torch_rdpg_forward_duckieboat.py" name="run_rdpg_duckieboat" output="screen">
            <param name="model" value="s1536_f1869509.pth" />
            <remap from="laser" to="RL/scan" />
            <remap from="odom" to="p3d_odom" />
            <remap from="cmd_vel" to="cmd_vel" />
            <remap from="joy" to="/joy" />
        </node>
    </group>

</launch>